script_editor/app/views/plays/show.html.erb

```diff
@@ -422,26 +457,40 @@
             down = toCut = false
         }
 
+        //next 2 functions actually change the styling of the items
         function cut(element) {
             if (element.className == "word" || element.className == "punc" ||
-                element.className == "stage") {
+                element.className.indexOf("stage") !== -1 || (element.className == "speaker" && speakerCut == true)) {
                     element.style.color = "#D3D3D3"
                     element.style.textDecoration = "line-through"
                     element.dataset.cut = "true"
                     element.style.backgroundColor = '#FFFFFF'
+                    speakerCut = false
             }
         }
 
         function uncut(element) {
             if (element.className == "word" || element.className == "punc" ||
-                element.className == "stage") {
+                element.className.indexOf("stage") !== -1 || (element.className == "speaker" && speakerCut == true)) {
                     element.style.color = "#000000"
                     element.style.textDecoration = "none"
                     element.dataset.cut = "false"
                     element.style.backgroundColor = '#FFFFFF'
+                    speakerCut = false
             }
         }
 
+        //obtains the index of the next speech, act, or scene which ends the speech to be cut
+        function nextSpeaker(srcElement, startIndex) {
+            items = Array.from(srcElement.parentNode.children).splice(startIndex)
+            for (element of items) {
+                if (element.className == "speaker" || element.className == "stage" || element.className == "sceneDiv" || element.className == "actDiv") {
```

Comment:
This isn't vital, but one way to avoid such a long series of ors is to make an array of the strings, like `array =["speaker", "stage", etc.]` and then check `array.includes(element.className )`